Lib/typing.py

```diff
@@ -3337,3 +3339,43 @@ def method(self) -> None:
         # read-only property, TypeError if it's a builtin class.
         pass
     return method
+
+
+def is_protocol(tp: type, /) -> bool:
+    """Return True if the given type is a Protocol.
+
+    Example::
+
+        >>> from typing import Protocol, is_protocol
+        >>> class P(Protocol):
+        ...     def a(self) -> str: ...
+        ...     b: int
+        >>> is_protocol(P)
+        True
+        >>> is_protocol(int)
+        False
+    """
+    return (
+        isinstance(tp, type)
+        and getattr(tp, '_is_protocol', False)
+        and tp is not Protocol
```

Comment:
Previously you said that we shouldn't do this, because you wanted this function to be compatible with `typing_extensions.Protocol` out of the box:

```suggestion
        issubclass(tp, Protocol)
        and getattr(tp, '_is_protocol', False)
        and tp is not Protocol
```

But I think `typing.is_protocol` won't work with `typing_extensions.Protocol` anyway, if we start reimplementing `Protocol` on 3.12 for whatever reason. `typing.is_protocol(typing_extensions.Protocol)` will evaluate to `True`, since `typing_extensions.Protocol is typing.Protocol` evaluates to `False`.

If you feel strongly about making this function work with `typing_extensions.Protocol` out of the box, you could do something like this, making use of the `typing_extensions._ProtocolMeta.__eq__` hack we have going at the moment:

```suggestion
        isinstance(tp, type)
        and getattr(tp, '_is_protocol', False)
        and tp != Protocol
```

But, I'm also not really sure how much considerations of third-party backport packages should be influencing our design of functions over at CPython. idk.

Reply:
I do think it's important to work well with typing-extensions, as its role is rather more central in the typing ecosystem than just a third-party backport package. Plus, wouldn't it be bad if we didn't use duck typing to decide whether something is a Protocol?

Reply:
> Plus, wouldn't it be bad if we didn't use duck typing to decide whether something is a Protocol?

For me it would be more like a delicious irony :p

But the patch looks good now, feel free to merge!